eng/common/templates/jobs/jobs.yml

```diff
@@ -43,6 +50,12 @@ jobs:
 
       name: ${{ job.job }}
 
+- ${{ if eq(parameters.runBuildFromSource, true) }}:
+  - template: /eng/templates/jobs/build-from-source.yml
```

Comment:
I think we need something in the dependency list to make sure source-build completes before publish to BAR:

https://github.com/dotnet/source-build-reference-packages/blob/89ccfa3d302356ce83125d6d5108662e18f0e2e4/eng/common/templates/jobs/jobs.yml#L61-L70

But I don't know a way to add all source-build jobs in a maintainable way without making this yaml even worse than my earlier suggestions. The list of source-build jobs only exists in the source-build template, so we'd need to rearrange stuff and make some extra templates to get it across... I can't stand that when we want to do the tiniest bit of data reuse, AzDO puts us on this convoluted path.

---

I think we can escape.... we can include a NOP/sentinel build job in our source-build template that has a consistent name and depends on every source-build job. Then in jobs.yml, we'd add a dependency on that sentinel job if source-build is enabled. Repos that don't use jobs.yml can simply use that sentinel job's consistent name. Waste of time, since AFAIK AzDO doesn't has server jobs yet that would let us avoid spinning up an actual machine to "do" the NOP. But at the moment it seems like the best we can do for reusability/maintainability.

To make it less wasteful for managed-only repos, we could use the sentinel job in a flexible way rather than *always* spinning up a specifically-named job for every target:
* Managed-only:
  * `source-build-main` (runs "centos71_portable" but uses the sentinel job name, so others can still add dependencies on it)
* Non-manged-only:
  * `source-build-centos71_portable`
  * `source-build-macOS_portable`
  * `source-build-centos71`
  *  ...,
  * `source-build-main` (does nothing, has dependencies on all previous jobs)

Reply:
> AFAIK AzDO doesn't has server jobs yet

Oh: https://docs.microsoft.com/en-us/azure/devops/pipelines/process/phases?view=azure-devops&tabs=yaml#server-jobs

Can probably keep it simple and always have one of those as the sentinel.

Reply:
Thanks for catching that.  I updated with a sentinel job for source-build and setup the dependency.  I tested it in my test repo and it seems to work great.  I also updated job/display names to better match the arcade standard for their AzDo jobs.